homeassistant/components/device_tracker/config_entry.py

```diff
@@ -0,0 +1,126 @@
+"""Code to set up a device tracker platform using a config entry."""
+from typing import Optional
+
+from homeassistant.helpers.typing import GPSType
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.const import (
+    STATE_NOT_HOME,
+    STATE_HOME,
+    ATTR_GPS_ACCURACY,
+    ATTR_LATITUDE,
+    ATTR_LONGITUDE,
+    ATTR_BATTERY_LEVEL,
+)
+from homeassistant.components import zone
+
+from .const import (
+    ATTR_SOURCE_TYPE,
+    DOMAIN,
+    LOGGER,
+    SOURCE_TYPE_GPS,
+)
+
+
+async def async_setup_entry(hass, entry):
+    """Set up an entry."""
+    component = hass.data.get(DOMAIN)  # type: Optional[EntityComponent]
+
+    if component is None:
+        component = hass.data[DOMAIN] = EntityComponent(
+            LOGGER, DOMAIN, hass
+        )
+
+    return await component.async_setup_entry(entry)
+
+
+async def async_unload_entry(hass, entry):
+    """Unload an entry."""
+    return await hass.data[DOMAIN].async_unload_entry(entry)
+
+
+class DeviceTrackerEntity(Entity):
```

Comment:
Will this mean that only config entry integrations can be migrated to the new entity?

Reply:
For this PR, yeah. For others, as we need to find a way to detect if they support entities or the see function.

Reply:
Currently all device trackers support the see function though, right? So if we want to avoid breaking that, all of them need to support that somehow. That doesn't mean they can't be new entities, or?

Reply:
We should be able to migrate the see function regardless of see service I think. The see function will schedule a state update.

Reply:
Because we don't have a solution yet for all legacy features like consider home etc, those cannot be implemented without a breaking change. Until now, all the other PRs have been non-breaking changes.

Reply:
It will be a breaking change to not support the see service.